lib/puma/cluster.rb

```diff
@@ -438,6 +438,10 @@ def run
                 when "p"
                   w.ping!(result.sub(/^\d+/,'').chomp)
                   @launcher.events.fire(:ping!, w)
+                  if !booted && @workers.none? {|worker| worker.last_status.empty?}
```

Comment:
this conditional kind of hurts my head but I don't have a better suggestion

Reply:
Extract logic to helper method with clever name?